operators/src/pro/cache/tile_cache.rs

```diff
@@ -389,15 +441,37 @@ impl TileCache {
     {
         let mut backend = self.backend.write().await;
 
-        let (entry_id, typed_stream) = {
-            let cache = backend.operator_caches.get(&key)?;
+        let (result, expired_entries) = {
+            // lookup the corresponding cache
+            let cache = backend.operator_caches.get_mut(&key)?;
+
+            let FindMatchResult {
+                match_found,
+                expired_entries,
+            } = cache.find_match(query);
+
+            let result = match_found.map(|entry_id| {
+                // if match was found, get the tile stream
+                (
+                    entry_id,
+                    cache
+                        .entries
+                        .get(&entry_id)
+                        .expect("entry should exist because it was returned from `find_match`")
+                        .tile_stream(query),
+                )
+            });
 
-            let (id, entry) = cache.find_match(query)?;
+            (result, expired_entries)
+        };
 
-            let typed_stream = entry.tile_stream(query);
+        // remove expired entries from lru index
+        for entry in expired_entries {
+            backend.lru.pop_entry(&entry);
+        }
 
-            (*id, typed_stream)
-        };
+        // unpack match if there is one
+        let (entry_id, typed_stream) = result?;
```

Comment:
When you do `?`, you could fail to promote. Is that ok?

Reply:
it only fails here if no match was found, so there is nothing to promote